core/src/plan/primary_key.rs

```diff
@@ -1,3 +1,13 @@
+//! Submodule providing the planner for handling primary keys.
+//!
+//! # Primary Key Planner
+//! The primary goal of the primary key planner is to optimize the query by using the primary key
+//! whenever possible. This is done by checking if the primary key is used in the query, for instance
+//! as part of the WHERE clause. If the primary key is used, the planner will remove the primary key
+//! from the WHERE clause and move it into the index field. This will allow the query to be executed
+//! more efficiently by calling directly the Store::fetch_data instead of the Store::scan_data.
+//!
+
```

Comment:
there looks that some edge cases still exist.
e.g.
```rust
let sql = "
    SELECT * FROM Player WHERE
        id = 100
        AND name = 'foo'
        AND (
            SELECT * FROM Player WHERE name = 'bar' AND Player.id = 1
        )
;";
```
in this case, the current planner cannot find primary key in subquery.

Reply:
Ok, please add a couple of tests to this case.

Reply:
I have managed to add a few tests and reproducing, now fixing the thing.

Reply:
Should now be handled by commit [71c4fc2](https://github.com/gluesql/gluesql/pull/1515/commits/71c4fc2db7d3f1a771a5bcff50764f5c291c82e7)